mapper/tedge_mapper/src/sm_c8y_mapper/mapper.rs

```diff
@@ -202,4 +219,192 @@ impl CumulocitySoftwareManagement {
             .await?;
         Ok(())
     }
+
+    async fn send_software_list_http(
+        &self,
+        json_response: &SoftwareListResponse,
+    ) -> Result<(), SMCumulocityMapperError> {
+        let token = get_jwt_token(&self.client).await?;
+
+        let reqwest_client = reqwest::ClientBuilder::new().build()?;
+
+        let url_host = self.config.query_string(C8yUrlSetting)?;
+        let url = get_url_for_sw_list(&url_host, &self.c8y_internal_id);
+
+        let c8y_software_list: C8yUpdateSoftwareListResponse = json_response.into();
+
+        let _published =
+            publish_software_list_http(&reqwest_client, &url, &token.token(), &c8y_software_list)
+                .await?;
+
+        Ok(())
+    }
+}
+
+async fn publish_software_list_http(
+    client: &reqwest::Client,
+    url: &str,
+    token: &str,
+    list: &C8yUpdateSoftwareListResponse,
+) -> Result<(), SMCumulocityMapperError> {
+    let request = client
+        .put(url)
+        .json(list)
+        .bearer_auth(token)
+        .timeout(Duration::from_millis(10000))
+        .build()?;
+
+    let _response = client.execute(request).await?;
+
+    Ok(())
+}
+
+async fn try_get_internal_id(
+    client: &reqwest::Client,
+    url_get_id: &str,
+    token: &str,
+) -> Result<String, SMCumulocityMapperError> {
+    let internal_id = client.get(url_get_id).bearer_auth(token).send().await?;
+
+    let internal_id_response = internal_id.json::<InternalIdResponse>().await?;
+
+    let internal_id = internal_id_response.id();
+    Ok(internal_id)
+}
+
+fn get_url_for_sw_list(url_host: &str, internal_id: &str) -> String {
+    let mut url_update_swlist = String::new();
+    url_update_swlist.push_str("https://");
+    url_update_swlist.push_str(url_host);
+    url_update_swlist.push_str("/inventory/managedObjects/");
+    url_update_swlist.push_str(internal_id);
+
+    url_update_swlist
+}
+
+fn get_url_for_get_id(url_host: &str, device_id: &str) -> String {
+    let mut url_get_id = String::new();
+    url_get_id.push_str("https://");
+    url_get_id.push_str(url_host);
+    url_get_id.push_str("/identity/externalIds/c8y_Serial/");
+    url_get_id.push_str(device_id);
+
+    url_get_id
+}
+
+async fn get_jwt_token(client: &Client) -> Result<SmartRestJwtResponse, SMCumulocityMapperError> {
+    let mut subscriber = client
+        .subscribe(Topic::new("c8y/s/dat").unwrap().filter())
+        .await?;
+
+    let () = client
+        .publish(mqtt_client::Message::new(
+            &"c8y/s/uat".into(),
+            "".to_string(),
+        ))
+        .await?;
+
+    let token_smartrest =
+        match tokio::time::timeout(Duration::from_secs(10), subscriber.next()).await {
+            Ok(Some(msg)) => msg.payload_str()?.to_string(),
+            Ok(None) => return Err(SMCumulocityMapperError::InvalidMqttMessage),
+            Err(err) => return Err(SMCumulocityMapperError::FromElapsed(err)),
+        };
```

Comment:
What if we receive some msg that is not a JwtToken (i.e. with a code that is not 71)?

* Is there a guaranty to receive only JwtToken on  `c8y/s/dat`?